interface/patient_file/encounter/Services/ViewHelper.php

```diff
@@ -0,0 +1,264 @@
+<?php
+/**
+ * Helper functions for an encounter view files
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License as published by the Free Software Foundation; either version 3 of the License, or (at your option)
+ * any later version. This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details. You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @subpackage Encounter
+ * @license GNU GPL3
+ * @author Robert Down <robertdown@live.com>
+ * @copyright Copyright (C) 2017 Robert Down
+ */
+
+namespace OpenEMR\Encounter\Services;
+
+class ViewHelper
+{
+
+    private $oldCategory = '';
+
+    static function getRegistry($state = "1", $limit = "unlimited", $offset = "0")
+    {
+        global $attendant_type;
+        $sql = "SELECT * FROM registry WHERE {column}= 1 AND state LIKE ? ORDER BY category, priority, name";
+        $params = array();
+        $column = ($attendant_type == 'pid') ? 'patient_encounter' : 'therapy_group_encounter';
+        $sql = str_replace("{column}", $column, $sql);
+        $params[1] = $state;
+        if ($limit != "unlimited") {
+            $params[2] = $limit;
+            $params[3] = $offset;
+            $sql .= " LIMIT ?, ?";
```

Comment:
I am pretty sure limit and offset can't use binding. We have the escape_limit() function to escape these:
https://github.com/openemr/openemr/blob/master/library/formdata.inc.php#L48
(note this library is included in all scripts, so can use it here)